alacritty_terminal/src/term/mod.rs

```diff
@@ -456,15 +465,33 @@ impl<'a, C> Iterator for RenderableCellsIter<'a, C> {
     #[inline]
     fn next(&mut self) -> Option<Self::Item> {
         loop {
-            if self.cursor_offset == self.inner.offset() && self.inner.column() == self.cursor.col {
-                let selected = self.is_selected(Point::new(self.cursor.line, self.cursor.col));
+            if self.cursor.point.line == self.inner.line()
+                && self.cursor.point.col == self.inner.column()
+            {
+                let selected = self.is_selected(self.cursor.point);
 
                 // Handle cursor
-                if let Some(cursor_key) = self.cursor_key.take() {
```

Comment:
you're already handling cursor here, so comment seems a bit out of place. I feel like it was wrongly moved at some point.